src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/SecretStorage/FileStorage.php

```diff
@@ -21,25 +21,30 @@ class FileStorage extends BaseStorage
     private $secretData = [];
 
     /**
-     * FileStorage constructor
+     * Initialize secret data value which represents encrypted credentials
+     *
+     * @return void
      * @throws TestFrameworkException
      */
-    public function __construct()
+    private function initialize(): void
```

Comment:
@karyna-tsymbal-atwix 
Can we try not to change the constructor to initialize()? Do you have some hard dependencies that are difficult to mock?

If it's difficult to find other solutions, please make sure initialization is done properly here:


```
private function initializeFileStorage()
    {
        // Initialize file storage
        try {
            $this->credStorage[self::ARRAY_KEY_FOR_FILE]  = new FileStorage();
        } catch (TestFrameworkException $e) {
            // Print error message in console
            print_r($e->getMessage());
            // Save to exception context for Allure report
            $this->setExceptionContexts(self::ARRAY_KEY_FOR_FILE, $e->getMessage());
        }
    }
```

Reply:
Hi @jilu1 , the problem here is that we need real private methods from the constructor to be executed (including parent constructor), but with our test credentials here https://github.com/magento/magento2-functional-testing-framework/pull/842/files#diff-ae8cfe83e67da69395e70368c54[…]25516f88e60225a94aa557060dda04L31

I didn't find a proper way to avoid calling a real constructor and execute parent constructor&private methods from constructor, at the same time.
Regarding your question about initializeFileStorage() method - I was checking it's usages, and I think it should be ok.
In the FileStorage class, an entry point is getEncryptedValue() method. I think it's the only case when we need those methods from constructor (which I moved to initialize()) to be executed.

Reply:
@karyna-tsymbal-atwix 
I am ok with using initialize() if there is no easy to mock private method in this case.

Practically getEncryptedValue() is the only entry point with the assumption that other public methods inherited from base class should be called after, but logically initialize() should be called right after FileStorage instantiation in initializeFileStorage() here https://github.com/magento/magento2-functional-testing-framework/blob/6042aea06dc6de1865577fcbc1596edf0861665d/src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/CredentialStore.php#L228.  I would add initialize() in both places.

Reply:
@jilu1  PR is updated with your suggestion, thank you